app/src/org/commcare/android/resource/installers/ProfileAndroidInstaller.java

```diff
@@ -97,6 +100,21 @@ public boolean install(Resource r, ResourceLocation location, Reference ref,
         return false;
     }
 
+    private void checkAppTarget() throws UnfullfilledRequirementsException {
+        SharedPreferences prefs = CommCareApp.currentSandbox.getAppPreferences();
+        if (prefs.contains(KEY_TARGET_PACKAGE_ID)) {
+            String targetPackage = prefs.getString(KEY_TARGET_PACKAGE_ID, "");
+            if (!StringUtils.isEmpty(targetPackage)) {
+                String myAppPackage = CommCareApplication.instance().getPackageName();
+                if (!myAppPackage.contentEquals(targetPackage)) {
+                    throw new UnfullfilledRequirementsException(
+                            "Your app doesn't match with this commcare flavour",
```

Comment:
This could also be a little more user-comprehensible. Maybe "This app requires [regular CommCare / CommCare LTS] to be installed"